src/v2/guide/typescript.md

```diff
@@ -187,3 +187,34 @@ const Component = Vue.extend({
 ```
 
 If you find type inference or member completion isn't working, annotating certain methods may help address these problems. Using the `--noImplicitAny` option will help find many of these unannotated methods.
+
+
+
+## Annotating Props
+
+```ts
+import Vue, { PropType } from 'vue'
+
+interface ComplexMessage { 
+  title: string,
+  okMessage: string,
+  cancelMessage: string
+}
+const Component = Vue.extend({
+  props: {
+    name: String,
+    success: { type: String },
+    callback: { 
+      type: Function as PropType<() => void>
+    },
+    message: {
+      type: Object as PropType<ComplexMessage>,
+      required: true,
+      validator (message: ComplexMessage) {
+        return !!message.title;
+      }
+    }
+  }
+})
+```
+If you find validator not getting type inference or member completion isn't working, annotating the argument the expected type may help address these problems.
```

Comment:
I think the sentence is broken. What exactly does it mean by "annotating the argument the expected type may help address these problems?"

Reply:
If you don't add the argument type to `validator`, typescript won't resolve the correct type for the prop, leading to not have autocompletion, for example this won't work as expected: 
```ts
  message: {
      type: Object as PropType<ComplexMessage>,
      required: true,
      validator (message) {
        return !!message.title;
      }
    } // typescript won't be able to resolve this type correctly
```

I admit the wording is not one of my strengths

Reply:
I think what you mean is "annotating the argument _with_ the expected type" then.